File: guess-the-number/guess-the-number-2.hs
import Control.Monad
import System.Exit
import System.Random
import Text.Printf

from, to :: Int
from = 1
to = 100

main :: IO ()
main = do target <- randomRIO (from, to)
          printf "Guess a number between %d and %d\n" from to
          forever $ do
            guess <- readLn
            case compare guess target of
              EQ -> do putStrLn "Correct! You got it!"
                       exitWith ExitSuccess
              LT -> putStrLn "higher"
              GT -> putStrLn "lower"

File: guess-the-number/guess-the-number-2.rb
from = 1
to = 100

target = rand from..to
puts "Guess a number between #{from} and #{to}"

loop do
  guess = gets.to_i

  case guess <=> target # 0 == EQ, -1 == LT, +1 == GT
  when 0
    puts "Correct! You got it!"
    break
  when -1 then puts "higher"
  when +1 then puts "lower"
  end
end
